src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -34,6 +34,21 @@ public function __construct(EntityManager $entityManager, ClassMetadata $class)
         $this->associationHydrator = new AssociationHydrator($entityManager, $class);
     }
 
+    public function findAggregate(mixed $id): ?ProductInterface
+    {
+        return $this->createQueryBuilder('o')
+            ->addSelect('values')
```

Comment:
Does `$this->associationHydrator` feature do the same as this technique or it's something else?

Reply:
To be honest, I'm not familiar with the association hydrator. The technique used by me forces the Doctrine mechanism to load the associations fully. It's like eager loading, but the eager doesn't support this direction of associations.

Reply:
Let's use hydrator then, especially it's already initialized on the repository.

Reply:
Guys, as the performance difference is negligible, I don't see any reason here to change this into hydration. The current solution generates only one query and forces the associations to be initialized.